smdebug/tensorflow/base_hook.py

```diff
@@ -401,6 +401,12 @@ def _get_collections_with_tensor(self, tf_tensor_name) -> Set["Collection"]:
         # matched with collections by regex
         if self.tape:
             return super()._get_collections_with_tensor(tf_tensor_name)
+        if (
+            tf_tensor_name not in self.tensor_to_collections
+            and is_tf_version_2x()
+            and tf.executing_eagerly()
+        ):
```

Comment:
can this if be merged with the one on L402?

Reply:
Done.